Unsupported data type Int64 for function floor

**Describe the bug**

error:

```
[ERROR] Query failed in native engine blaze: Job aborted due to stage failure: Task 0 in stage 8073.0 failed 1 times, most recent failure: Lost task 0.0 in stage 8073.0 (TID 5606) (fv-az1945-468.r3m3350mam0ergycvke4vbykda.phxx.internal.cloudapp.net executor driver): java.lang.RuntimeException: poll record batch error: Execution error: native execution panics: Execution error: Execution error: output_with_sender[Project] error: Execution error: output_with_sender[Project]: output() returns error: Execution error: Unsupported data type Int64 for function floor
	at org.apache.spark.sql.blaze.JniBridge.nextBatch(Native Method)
	at org.apache.spark.sql.blaze.BlazeCallNativeWrapper$$anon$1.hasNext(BlazeCallNativeWrapper.scala:80)
	at org.apache.spark.util.CompletionIterator.hasNext(CompletionIterator.scala:31)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
	at org.apache.spark.util.random.SamplingUtils$.reservoirSampleAndCount(SamplingUtils.scala:41)
	at org.apache.spark.RangePartitioner$.$anonfun$sketch$1(Partitioner.scala:322)
	at org.apache.spark.RangePartitioner$.$anonfun$sketch$1$adapted(Partitioner.scala:320)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsWithIndex$2(RDD.scala:910)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsWithIndex$2$adapted(RDD.scala:910)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
```

**To Reproduce**

```
create table t1 using parquet as select 1L as c1;
select floor(c1), from t1;
```

**Expected behavior**


**Screenshots**


**Additional context**

--- a/spark-extension-shims-spark3/src/test/scala/org/apache/spark/sql/blaze/BlazeQuerySuite.scala
+++ b/spark-extension-shims-spark3/src/test/scala/org/apache/spark/sql/blaze/BlazeQuerySuite.scala
@@ -86,19 +86,24 @@ class BlazeQuerySuite extends org.apache.spark.sql.QueryTest with BaseBlazeSQLSu
       sql("create table t1(c1 binary, c2 int) using parquet")
       sql(
         "insert into t1 values (to_binary('test1', 'utf-8'), 1), (to_binary('test2', 'utf-8'), 2)")
-      spark.table("t1").printSchema()
       val df = sql("select c2 from t1 order by c1")
       checkAnswer(df, Seq(Row(1), Row(2)))
     }
   }
-  
+
   test("log function with negative input") {
     withTable("t1") {
       sql("create table t1 using parquet as select -1 as c1")
-      spark.table("t1").printSchema()
       val df = sql("select ln(c1) from t1")
-      df.show()
       checkAnswer(df, Seq(Row(null)))
+    }
+  }
+
+  test("floor function with long input") {
+    withTable("t1") {
+      sql("create table t1 using parquet as select 1L as c1, 2.2 as c2")
+      val df = sql("select floor(c1), floor(c2) from t1")
+      checkAnswer(df, Seq(Row(1, 2)))
     }
   }
 }

--- a/spark-extension/src/main/scala/org/apache/spark/sql/blaze/NativeConverters.scala
+++ b/spark-extension/src/main/scala/org/apache/spark/sql/blaze/NativeConverters.scala
@@ -827,13 +827,17 @@ object NativeConverters extends Logging {
       case e: Log10 =>
         buildScalarFunction(pb.ScalarFunction.Log10, e.children.map(nullIfNegative), e.dataType)
       case e: Floor if !e.dataType.isInstanceOf[DecimalType] =>
-        buildExprNode {
-          _.setTryCast(
-            pb.PhysicalTryCastNode
-              .newBuilder()
-              .setExpr(buildScalarFunction(pb.ScalarFunction.Floor, e.children, e.dataType))
-              .setArrowType(convertDataType(e.dataType))
-              .build())
+        if (e.child.dataType.isInstanceOf[LongType]) {
+          convertExprWithFallback(e.child, isPruningExpr, fallback)
+        } else {
+          buildExprNode {
+            _.setTryCast(
+              pb.PhysicalTryCastNode
+                .newBuilder()
+                .setExpr(buildScalarFunction(pb.ScalarFunction.Floor, e.children, e.dataType))
+                .setArrowType(convertDataType(e.dataType))
+                .build())
+          }
         }
       case e: Ceil if !e.dataType.isInstanceOf[DecimalType] =>
         buildExprNode {

--- a/spark-extension/src/main/scala/org/apache/spark/sql/execution/blaze/arrowio/ArrowFFIExporter.scala
+++ b/spark-extension/src/main/scala/org/apache/spark/sql/execution/blaze/arrowio/ArrowFFIExporter.scala
@@ -67,13 +67,11 @@ class ArrowFFIExporter(rowIter: Iterator[InternalRow], schema: StructType) {
       callRowIter(exportArrowArrayPtr)
     } else {
       // otherwise, process rows as native user
-      nativeCurrentUser.doAs(
-        new PrivilegedExceptionAction[Boolean] {
-          override def run(): Boolean = {
-            callRowIter(exportArrowArrayPtr)
-          }
+      nativeCurrentUser.doAs(new PrivilegedExceptionAction[Boolean] {
+        override def run(): Boolean = {
+          callRowIter(exportArrowArrayPtr)
         }
-      )
+      })
     }
   }
 

